aptos-move/framework/aptos-framework/sources/stake.move

```diff
@@ -1161,8 +1163,23 @@ module aptos_framework::stake {
             assume cur_validator_perf.successful_proposals + cur_validator_perf.failed_proposals <= MAX_U64;
         };
         let num_total_proposals = cur_validator_perf.successful_proposals + cur_validator_perf.failed_proposals;
-
-        let (rewards_rate, rewards_rate_denominator) = staking_config::get_reward_rate(staking_config);
+        let (rewards_rate, rewards_rate_denominator) = if (features::reward_rate_decrease_enabled()) {
```

Comment:
Why not just update distribute_rewards to take FixedPoint64?

Reply:
Good question. Actually I tried to use FixedPoint64 in `distribute_rewards`. Then many test cases are broken. I realized the nominator/denominator way has different precision from the FixedPoint64 way. I don't want to change the behavior of the existing logic so I keep `distribute_rewards` untouched.

Reply:
Fixed point cannot represent some rational numbers without error. For example, if I want 1/5=0.2, the closest value of a FixedPoint32 could represent is 0.19999999995343387. FixedPoint64 can only make the error smaller but it cannot eliminate the error. But the error is really small so I think it's fine.